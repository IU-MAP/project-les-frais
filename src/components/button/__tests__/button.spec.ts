import { mount, shallowMount } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router';
import Button from '../index.vue';

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      path: '/',
      component: {
        template: 'Welcome to the Les Frais',
      },
    },
    {
      path: '/login',
      component: {
        template: 'Login',
      },
    },
  ],
});

describe('Button', () => {
  describe('when link is not provided', () => {
    it('matches the snapshot', async () => {
      router.push('/');
      await router.isReady();
      const look = 'default';
      const wrapper = mount(Button, {
        props: { look },
        global: {
          plugins: [router],
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  
    it('emits event on click', async () => {
      router.push('/');
      await router.isReady();
      const look = 'default';
      const wrapper = mount(Button, {
        props: { look },
        global: {
          plugins: [router],
        },
      });
      wrapper.trigger('click');
      expect(wrapper.emitted()).toHaveProperty('click');
    });
  });

  describe('when link id provided', () => {
    const conf = {
      props: { look: 'default', link: '/login' },
      global: {
        plugins: [router],
      },
    };
    it('matches the snapshot', async () => {
      router.push('/');
      await router.isReady();
      const wrapper = mount(Button, conf);
      expect(wrapper).toMatchSnapshot();
    });

    it('has the link from the props', async () => {
      router.push('/');
      await router.isReady();
      const wrapper = shallowMount(Button, conf);
      expect(wrapper.html()).toContain('to="/login"');
    });
  });
});
