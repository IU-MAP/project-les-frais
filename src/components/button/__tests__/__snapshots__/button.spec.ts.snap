// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button when link id provided matches the snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <a
        class="btn default"
        href="/login"
      >
        
        
      </a>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "RouterLink": Object {
          "__emits": null,
          "__props": Array [
            Object {
              "activeClass": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "ariaCurrentValue": Object {
                "0": false,
                "1": true,
                "default": "page",
                "type": [Function],
              },
              "custom": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
              "exactActiveClass": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "replace": Object {
                "0": true,
                "1": true,
                "type": [Function],
              },
              "to": Object {
                "0": false,
                "1": true,
                "required": true,
                "type": Array [
                  [Function],
                  [Function],
                ],
              },
            },
            Array [
              "replace",
              "custom",
              "ariaCurrentValue",
            ],
          ],
          "name": "RouterLink",
          "props": Object {
            "activeClass": [Function],
            "ariaCurrentValue": Object {
              "0": false,
              "1": true,
              "default": "page",
              "type": [Function],
            },
            "custom": [Function],
            "exactActiveClass": [Function],
            "replace": [Function],
            "to": Object {
              "0": false,
              "1": true,
              "required": true,
              "type": Array [
                [Function],
                [Function],
              ],
            },
          },
          "setup": [Function],
        },
        "RouterView": Object {
          "inheritAttrs": false,
          "name": "RouterView",
          "props": Object {
            "name": Object {
              "default": "default",
              "type": [Function],
            },
            "route": [Function],
          },
          "setup": [Function],
        },
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$router": Object {
            "addRoute": [Function],
            "afterEach": [Function],
            "back": [Function],
            "beforeEach": [Function],
            "beforeResolve": [Function],
            "currentRoute": RefImpl {
              "__v_isRef": true,
              "_rawValue": Object {
                "fullPath": "/",
                "hash": "",
                "href": "/",
                "matched": Array [
                  Object {
                    "aliasOf": undefined,
                    "beforeEnter": undefined,
                    "children": Array [],
                    "components": Object {
                      "default": Object {
                        "template": "Welcome to the Les Frais",
                      },
                    },
                    "enterCallbacks": Object {},
                    "instances": Object {},
                    "leaveGuards": Set {},
                    "meta": Object {},
                    "name": undefined,
                    "path": "/",
                    "props": Object {
                      "default": false,
                    },
                    "redirect": undefined,
                    "updateGuards": Set {},
                  },
                ],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
              "_shallow": true,
              "_value": Object {
                "fullPath": "/",
                "hash": "",
                "href": "/",
                "matched": Array [
                  Object {
                    "aliasOf": undefined,
                    "beforeEnter": undefined,
                    "children": Array [],
                    "components": Object {
                      "default": Object {
                        "template": "Welcome to the Les Frais",
                      },
                    },
                    "enterCallbacks": Object {},
                    "instances": Object {},
                    "leaveGuards": Set {},
                    "meta": Object {},
                    "name": undefined,
                    "path": "/",
                    "props": Object {
                      "default": false,
                    },
                    "redirect": undefined,
                    "updateGuards": Set {},
                  },
                ],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
            },
            "forward": [Function],
            "getRoutes": [Function],
            "go": [Function],
            "hasRoute": [Function],
            "install": [Function],
            "isReady": [Function],
            "onError": [Function],
            "options": Object {
              "history": Object {
                "base": "",
                "createHref": [Function],
                "destroy": [Function],
                "go": [Function],
                "listen": [Function],
                "location": "/",
                "pauseListeners": [Function],
                "push": [Function],
                "replace": [Function],
                "state": Object {
                  "back": null,
                  "current": "/",
                  "forward": null,
                  "position": 0,
                  "replaced": true,
                  "scroll": null,
                },
              },
              "routes": Array [
                Object {
                  "component": Object {
                    "template": "Welcome to the Les Frais",
                  },
                  "path": "/",
                },
                Object {
                  "component": Object {
                    "template": "Login",
                  },
                  "path": "/login",
                },
              ],
            },
            "push": [Function],
            "removeRoute": [Function],
            "replace": [Function],
            "resolve": [Function],
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        Symbol([vue-router]: router): Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "/",
              "pauseListeners": [Function],
              "push": [Function],
              "replace": [Function],
              "state": Object {
                "back": null,
                "current": "/",
                "forward": null,
                "position": 0,
                "replaced": true,
                "scroll": null,
              },
            },
            "routes": Array [
              Object {
                "component": Object {
                  "template": "Welcome to the Les Frais",
                },
                "path": "/",
              },
              Object {
                "component": Object {
                  "template": "Login",
                },
                "path": "/login",
              },
            ],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
        Symbol([vue-router]: route location): Object {
          "fullPath": "/",
          "hash": "",
          "matched": Array [
            Object {
              "aliasOf": undefined,
              "beforeEnter": undefined,
              "children": Array [],
              "components": Object {
                "default": Object {
                  "template": "Welcome to the Les Frais",
                },
              },
              "enterCallbacks": Object {},
              "instances": Object {},
              "leaveGuards": Set {},
              "meta": Object {},
              "name": undefined,
              "path": "/",
              "props": Object {
                "default": false,
              },
              "redirect": undefined,
              "updateGuards": Set {},
            },
          ],
          "meta": Object {},
          "name": undefined,
          "params": Object {},
          "path": "/",
          "query": Object {},
          "redirectedFrom": undefined,
        },
        Symbol([vue-router]: router view location): RefImpl {
          "__v_isRef": true,
          "_rawValue": Object {
            "fullPath": "/",
            "hash": "",
            "href": "/",
            "matched": Array [
              Object {
                "aliasOf": undefined,
                "beforeEnter": undefined,
                "children": Array [],
                "components": Object {
                  "default": Object {
                    "template": "Welcome to the Les Frais",
                  },
                },
                "enterCallbacks": Object {},
                "instances": Object {},
                "leaveGuards": Set {},
                "meta": Object {},
                "name": undefined,
                "path": "/",
                "props": Object {
                  "default": false,
                },
                "redirect": undefined,
                "updateGuards": Set {},
              },
            ],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
          "_shallow": true,
          "_value": Object {
            "fullPath": "/",
            "hash": "",
            "href": "/",
            "matched": Array [
              Object {
                "aliasOf": undefined,
                "beforeEnter": undefined,
                "children": Array [],
                "components": Object {
                  "default": Object {
                    "template": "Welcome to the Les Frais",
                  },
                },
                "enterCallbacks": Object {},
                "instances": Object {},
                "leaveGuards": Set {},
                "meta": Object {},
                "name": undefined,
                "path": "/",
                "props": Object {
                  "default": false,
                },
                "redirect": undefined,
                "updateGuards": Set {},
              },
            ],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "_uid": 2,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$router": Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "/",
              "pauseListeners": [Function],
              "push": [Function],
              "replace": [Function],
              "state": Object {
                "back": null,
                "current": "/",
                "forward": null,
                "position": 0,
                "replaced": true,
                "scroll": null,
              },
            },
            "routes": Array [
              Object {
                "component": Object {
                  "template": "Welcome to the Les Frais",
                },
                "path": "/",
              },
              Object {
                "component": Object {
                  "template": "Login",
                },
                "path": "/login",
              },
            ],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "clk": [Function],
    "hasOwnProperty": [Function],
    "isSubmit": false,
    "link": "/login",
    "look": "default",
  },
  "rootVM": Object {},
  "wrapperElement": <a
    class="btn default"
    href="/login"
  >
    
    
  </a>,
}
`;

exports[`Button when link is not provided matches the snapshot 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": null,
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <button
        class="btn default"
        type="button"
      >
        
        
      </button>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "RouterLink": Object {
          "name": "RouterLink",
          "props": Object {
            "activeClass": [Function],
            "ariaCurrentValue": Object {
              "default": "page",
              "type": [Function],
            },
            "custom": [Function],
            "exactActiveClass": [Function],
            "replace": [Function],
            "to": Object {
              "required": true,
              "type": Array [
                [Function],
                [Function],
              ],
            },
          },
          "setup": [Function],
        },
        "RouterView": Object {
          "inheritAttrs": false,
          "name": "RouterView",
          "props": Object {
            "name": Object {
              "default": "default",
              "type": [Function],
            },
            "route": [Function],
          },
          "setup": [Function],
        },
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$router": Object {
            "addRoute": [Function],
            "afterEach": [Function],
            "back": [Function],
            "beforeEach": [Function],
            "beforeResolve": [Function],
            "currentRoute": RefImpl {
              "__v_isRef": true,
              "_rawValue": Object {
                "fullPath": "/",
                "hash": "",
                "href": "/",
                "matched": Array [
                  Object {
                    "aliasOf": undefined,
                    "beforeEnter": undefined,
                    "children": Array [],
                    "components": Object {
                      "default": Object {
                        "template": "Welcome to the Les Frais",
                      },
                    },
                    "enterCallbacks": Object {},
                    "instances": Object {},
                    "leaveGuards": Set {},
                    "meta": Object {},
                    "name": undefined,
                    "path": "/",
                    "props": Object {
                      "default": false,
                    },
                    "redirect": undefined,
                    "updateGuards": Set {},
                  },
                ],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
              "_shallow": true,
              "_value": Object {
                "fullPath": "/",
                "hash": "",
                "href": "/",
                "matched": Array [
                  Object {
                    "aliasOf": undefined,
                    "beforeEnter": undefined,
                    "children": Array [],
                    "components": Object {
                      "default": Object {
                        "template": "Welcome to the Les Frais",
                      },
                    },
                    "enterCallbacks": Object {},
                    "instances": Object {},
                    "leaveGuards": Set {},
                    "meta": Object {},
                    "name": undefined,
                    "path": "/",
                    "props": Object {
                      "default": false,
                    },
                    "redirect": undefined,
                    "updateGuards": Set {},
                  },
                ],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
            },
            "forward": [Function],
            "getRoutes": [Function],
            "go": [Function],
            "hasRoute": [Function],
            "install": [Function],
            "isReady": [Function],
            "onError": [Function],
            "options": Object {
              "history": Object {
                "base": "",
                "createHref": [Function],
                "destroy": [Function],
                "go": [Function],
                "listen": [Function],
                "location": "/",
                "pauseListeners": [Function],
                "push": [Function],
                "replace": [Function],
                "state": Object {
                  "back": null,
                  "current": "/",
                  "forward": null,
                  "position": 0,
                  "replaced": true,
                  "scroll": null,
                },
              },
              "routes": Array [
                Object {
                  "component": Object {
                    "template": "Welcome to the Les Frais",
                  },
                  "path": "/",
                },
                Object {
                  "component": Object {
                    "template": "Login",
                  },
                  "path": "/login",
                },
              ],
            },
            "push": [Function],
            "removeRoute": [Function],
            "replace": [Function],
            "resolve": [Function],
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        Symbol([vue-router]: router): Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "/",
              "pauseListeners": [Function],
              "push": [Function],
              "replace": [Function],
              "state": Object {
                "back": null,
                "current": "/",
                "forward": null,
                "position": 0,
                "replaced": true,
                "scroll": null,
              },
            },
            "routes": Array [
              Object {
                "component": Object {
                  "template": "Welcome to the Les Frais",
                },
                "path": "/",
              },
              Object {
                "component": Object {
                  "template": "Login",
                },
                "path": "/login",
              },
            ],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
        Symbol([vue-router]: route location): Object {
          "fullPath": "/",
          "hash": "",
          "matched": Array [
            Object {
              "aliasOf": undefined,
              "beforeEnter": undefined,
              "children": Array [],
              "components": Object {
                "default": Object {
                  "template": "Welcome to the Les Frais",
                },
              },
              "enterCallbacks": Object {},
              "instances": Object {},
              "leaveGuards": Set {},
              "meta": Object {},
              "name": undefined,
              "path": "/",
              "props": Object {
                "default": false,
              },
              "redirect": undefined,
              "updateGuards": Set {},
            },
          ],
          "meta": Object {},
          "name": undefined,
          "params": Object {},
          "path": "/",
          "query": Object {},
          "redirectedFrom": undefined,
        },
        Symbol([vue-router]: router view location): RefImpl {
          "__v_isRef": true,
          "_rawValue": Object {
            "fullPath": "/",
            "hash": "",
            "href": "/",
            "matched": Array [
              Object {
                "aliasOf": undefined,
                "beforeEnter": undefined,
                "children": Array [],
                "components": Object {
                  "default": Object {
                    "template": "Welcome to the Les Frais",
                  },
                },
                "enterCallbacks": Object {},
                "instances": Object {},
                "leaveGuards": Set {},
                "meta": Object {},
                "name": undefined,
                "path": "/",
                "props": Object {
                  "default": false,
                },
                "redirect": undefined,
                "updateGuards": Set {},
              },
            ],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
          "_shallow": true,
          "_value": Object {
            "fullPath": "/",
            "hash": "",
            "href": "/",
            "matched": Array [
              Object {
                "aliasOf": undefined,
                "beforeEnter": undefined,
                "children": Array [],
                "components": Object {
                  "default": Object {
                    "template": "Welcome to the Les Frais",
                  },
                },
                "enterCallbacks": Object {},
                "instances": Object {},
                "leaveGuards": Set {},
                "meta": Object {},
                "name": undefined,
                "path": "/",
                "props": Object {
                  "default": false,
                },
                "redirect": undefined,
                "updateGuards": Set {},
              },
            ],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$router": Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "href": "/",
              "matched": Array [
                Object {
                  "aliasOf": undefined,
                  "beforeEnter": undefined,
                  "children": Array [],
                  "components": Object {
                    "default": Object {
                      "template": "Welcome to the Les Frais",
                    },
                  },
                  "enterCallbacks": Object {},
                  "instances": Object {},
                  "leaveGuards": Set {},
                  "meta": Object {},
                  "name": undefined,
                  "path": "/",
                  "props": Object {
                    "default": false,
                  },
                  "redirect": undefined,
                  "updateGuards": Set {},
                },
              ],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "/",
              "pauseListeners": [Function],
              "push": [Function],
              "replace": [Function],
              "state": Object {
                "back": null,
                "current": "/",
                "forward": null,
                "position": 0,
                "replaced": true,
                "scroll": null,
              },
            },
            "routes": Array [
              Object {
                "component": Object {
                  "template": "Welcome to the Les Frais",
                },
                "path": "/",
              },
              Object {
                "component": Object {
                  "template": "Login",
                },
                "path": "/login",
              },
            ],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.11",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "clk": [Function],
    "hasOwnProperty": [Function],
    "isSubmit": false,
    "link": null,
    "look": "default",
  },
  "rootVM": Object {},
  "wrapperElement": <button
    class="btn default"
    type="button"
  >
    
    
  </button>,
}
`;
